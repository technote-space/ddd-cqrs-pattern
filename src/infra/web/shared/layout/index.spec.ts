import type { ILoadingComponent } from '$/web/shared/loading';
import renderer from 'react-test-renderer';
import { LayoutComponent } from '.';

describe('LayoutComponent', () => {
  it('Layoutのコンテキストプロバイダー', () => {
    const mockRender = jest.fn(() => 'LoadingComponent');
    const component = new LayoutComponent({ render: mockRender } as never as ILoadingComponent);

    const tree = renderer.create(component.render({ children: 'Children' }));
    expect(tree).toMatchSnapshot();
  });
});
