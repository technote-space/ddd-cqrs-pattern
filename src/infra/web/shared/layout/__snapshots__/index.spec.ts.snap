// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent Layoutのコンテキストプロバイダー 1`] = `
<main>
  <div>
    Children
    LoadingComponent
  </div>
</main>
`;
